aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -217,14 +217,24 @@ Tensor empty_like(
                 "It is currently not supported to specify a dtype that doesn't match "
                 "the input tensor's dtype via empty_like.  Specified: ", options.dtype(),
                 " Input tensor's dtype: ", self.dtype());
-    // TODO: uncomment when qscheme diff is landed
-    // TORCH_INTERNAL_ASSERT(self.qscheme(), at::kPerTensorAffine,
-    //                       "empty_like for quantized Tensor only works for
-    //                        PerTensorAffine scheme right now");
-    return at::_empty_affine_quantized(self.sizes(), options,
-                                       self.q_scale(),
-                                       self.q_zero_point(),
-                                       use_memory_format);
+    auto qscheme = self.qscheme();
+    if (qscheme == kPerTensorAffine) {
+      return at::_empty_affine_quantized(self.sizes(), options,
```

Comment:
we have different ordering of arguments here...
maybe we can have:
```at::_empty_affine_quantized(q_scale, q_zero_point, sizes, options, use_memory_format);```
and
```at::_empty_per_channel_affine_quantized_like(q_per_channel_scales, q_per_channel_zero_points, q_per_channel_axis, sizes, options, use_memory_format);```

But should do in a separate PR: @dskhudia could you change the order for `_empty_per_channel_affine_quantized_like`?

Reply:
oh `q_per_channel_axis` is added in this PR, then @dzhulgakov could you change the order as well?

Reply:
looks like this is addressed in the later PRs

Reply:
So the order and name for `_empty_per_channel_affine_quantized_like` (_like!) - was weird as a workaround for some codegen peculiarity - https://github.com/pytorch/pytorch/pull/26243 fixes it

I the order is still not the same - TensorOptions and additional arguments are in the wrong order. It's C++ only as the args are marked as kwargs in python. But we should fix it anyway. Since it's an intrusive change - I'll send a separate PR for it after these ones land